gbdk-lib/examples/cross-platform/platformer_template/src/player.c

```diff
@@ -0,0 +1,355 @@
+#pragma bank 255
+
+#include <gbdk/platform.h>
+#include <gbdk/metasprites.h>
+#include <stdint.h>
+#include "common.h"
+#include "PlayerCharacterLeft.h"
+#include "PlayerCharacterRight.h"
+#include "camera.h"
+#include "level.h"
+
+BANKREF_EXTERN(PlayerCharacterLeft)
+BANKREF_EXTERN(PlayerCharacterRight)
+
+
+#define GRAVTY 15
+#define GROUND_FRICTION 15
+#define PLAYER_CHARACTER_INCREASE_JUMP_TIMER_MAX 20
+#define PLAYER_CHARACTER_JUMP_VELOCITY 450
+#define PLAYER_CHARACTER_WALK_VELOCITY 225
+#define PLAYER_CHARACTER_RUN_VELOCITY 325
+#define PLAYER_CHARACTER_WALK_TWO_FRAME_COUNTER 3
+#define PLAYER_CHARACTER_RUN_TWO_FRAME_COUNTER 5
+
+#define PLAYER_CHARACTER_METASPRITE_PIVOT_X 12
+#define PLAYER_CHARACTER_METASPRITE_PIVOT_Y 6
+#define PLAYER_CHARACTER_BOUNDING_BOX_HALF_WIDTH 5
+#define PLAYER_CHARACTER_BOUNDING_BOX_HALF_HEIGHT 12
+#define PLAYER_CHARACTER_BOUNDING_BOX_HEIGHT 24
+
+uint8_t facingRight =TRUE;
+
+uint8_t playerJumpIncrease = 0;
+uint8_t threeFrameCounter=0;
+uint16_t playerX, playerY;
+int16_t playerXVelocity, playerYVelocity;
+
+    // Flip horizontally, if we aren't facing right
+
+#if defined(NINTENDO_NES)
+    const uint8_t baseProp=4;
```

Comment:
I don't think this baseProp parameter is needed here, as it probably doesn't do what you want anyway.

The baseProp parameter to move_metasprite_ex will be added to the properties of the metasprite. However, adding 4 to it seems kind of pointless as that represents bit 2, and that bit doesn't even exist in the PPUs OAM memory. (is always zero after OAM DMA)

Furthermore, as you already have the different directions for the player as separate .png files, flipping the player here is redundant, even if you used the correct bit.